### PR TITLE
Teach x-ci-verify-versions to check that versions exist in the database

### PR DESCRIPTION
**Pull request by BillyONeal**
Initial results are fixed in https://github.com/microsoft/vcpkg/pull/33948

As part of adding these checks, I also substantially overhauled how x-ci-verify-versions works:
* Print error messages naming the file we're complaining about so that IDEs etc. can go to the file, (Using the standard path: kind: message format)
* Record the location in try_load_port and friends instead of requiring callers recover that themselves.
* Make load_git_versions_file return the expected path of the versions file so that callers need not recover that themselves.
* Deduplicate 'try vcpkg.json, also try CONTROL' behavior from x-add-version and x-ci-verify-versions.
* Don't stop validating versions information at the first error.

The remaining places that are still thinking about adding / "vcpkg.json" are now only:

* format-manifest when deciding the new path after parsing a CONTROL
* new when deciding the new path
* inside try_load_port and friends themselves
* the horrible mess that is how vcpkgpaths loads the consumer-manifest

Before and after output with an intentionally damaged repo follows.

Before:

```
PS C:\Dev\vcpkg> .\vcpkg.exe x-ci-verify-versions
FAIL: adios2
FAIL: boost
FAIL: cpprestsdk
FAIL: curl
FAIL: zlib
error: Found the following errors:
error: curl is missing a version database file at C:\Dev\vcpkg\versions\c-\curl.json
Run:
vcpkg x-add-version curl
to create the versions file.
error: while attempting to load local port cpprestsdk
C:\Dev\vcpkg\ports\cpprestsdk\vcpkg.json:8:7: error: Unexpected character in middle of array
  on expression:       "platform": "!uwp & !windows"
                       ^
error: while parsing versions for adios2 from C:\Dev\vcpkg\versions\a-\adios2.json
2.8.3#1 is declared with fac4ec886b3d46f0a48ee44988fc8224bce59ad7, but the local port has a different SHA 73405fc9493db67ddbd3ac578e41f01eea80df8c.
Please update the port's version fields and then run:
vcpkg x-add-version adios2
git add versions
git commit -m "Update version database"
to add the new version.
error: while parsing versions for boost from C:\Dev\vcpkg\versions\b-\boost.json
1.83.0#1 is declared with 315139e076ff5ea5672a158688934b04d2b7a815, but the local port has a different SHA 28c60c17f76c5dc525b0c64f7effdc2336ccb81b.
Please update the port's version fields and then run:
vcpkg x-add-version boost
git add versions
git commit -m "Update version database"
to add the new version.
error: while parsing versions for zlib from C:\Dev\vcpkg\versions\z-\zlib.json
1.3#2 is declared with 5ac18c6e6e3e2bf2a9e3d0bc8a845f198e4c4e05, but the local port has a different SHA a4344974697f4860565e4c5ba39d6b3e8086dbef.
Please update the port's version fields and then run:
vcpkg x-add-version zlib
git add versions
git commit -m "Update version database"
to add the new version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all
```

After:
```
PS C:\Dev\vcpkg> C:\Dev\vcpkg-tool\out\build\Win-x64-Debug-WithArtifacts\vcpkg.exe x-ci-verify-versions
C:\Dev\vcpkg\ports\cpprestsdk\vcpkg.json:8:7: error: Unexpected character in middle of array
  on expression:       "platform": "!uwp & !windows"
                       ^
C:\Dev\vcpkg\versions\a-\adios2.json: error: adios2@2.8.3#1 git tree fac4ec886b3d46f0a48ee44988fc8224bce59ad7 does not match the port directory
C:\Dev\vcpkg\ports\adios2: note: the port directory has git tree 73405fc9493db67ddbd3ac578e41f01eea80df8c
C:\Dev\vcpkg\ports\adios2\vcpkg.json: note: if adios2@2.8.3#1 is already published, update this file with a new version or port-version, commit it, then add the new version by running:
  vcpkg x-add-version adios2
  git add versions
  git commit -m "Update version database"
note: if adios2@2.8.3#1 is not yet published, overwrite the previous git tree by running:
  vcpkg x-add-version adios2 --overwrite-version
  git add versions
  git commit -m "Update version database"
C:\Dev\vcpkg\ports\adios2\vcpkg.json: error: the override of bzip2 names version 1.0#1 which does not exist in the version database. Installing this port at the top level will fail as that version will be unresolvable.
C:\Dev\vcpkg\versions\b-\bzip2.json: note: consider removing the version override or choosing a value declared here
C:\Dev\vcpkg\ports\aliyun-oss-c-sdk\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\aws-lambda-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\aws-sdk-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\azure-c-shared-utility\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\azure-core-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\versions\b-\boost.json: error: boost@1.83.0#1 git tree 315139e076ff5ea5672a158688934b04d2b7a815 does not match the port directory
C:\Dev\vcpkg\ports\boost: note: the port directory has git tree 28c60c17f76c5dc525b0c64f7effdc2336ccb81b
C:\Dev\vcpkg\ports\boost\vcpkg.json: note: if boost@1.83.0#1 is already published, update this file with a new version or port-version, commit it, then add the new version by running:
  vcpkg x-add-version boost
  git add versions
  git commit -m "Update version database"
note: if boost@1.83.0#1 is not yet published, overwrite the previous git tree by running:
  vcpkg x-add-version boost --overwrite-version
  git add versions
  git commit -m "Update version database"
C:\Dev\vcpkg\ports\cfitsio\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\ports\cocoyaxi\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named libcurl
C:\Dev\vcpkg\ports\cpr\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\cpr\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named ssl
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: this port is not in the version database
C:\Dev\vcpkg\versions\c-\curl.json: note: the version database file should be here
note: run 'vcpkg x-add-version curl' to create the version database file
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named http2
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named idn
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named idn
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named schannel
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named ssh
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named ssh
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named ssl
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named ssl
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named ssl
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named winldap
C:\Dev\vcpkg\ports\curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named winssl
C:\Dev\vcpkg\ports\curlpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\czmq\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\ports\date\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named remote-api
C:\Dev\vcpkg\ports\gdal\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\ports\google-cloud-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named rest-common
C:\Dev\vcpkg\ports\gz-fuel-tools8\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\idevicerestore\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\ignition-fuel-tools1\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\ignition-fuel-tools4\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\kinectsdk1\vcpkg.json: error: the "version>=" constraint to vcpkg-tool-lessmsi names version 1.10#1 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.C:\Dev\vcpkg\versions\v-\vcpkg-tool-lessmsi.json: note: consider removing the version constraint or choosing a value declared here
C:\Dev\vcpkg\ports\kinectsdk2\vcpkg.json: error: the "version>=" constraint to vcpkg-tool-lessmsi names version 1.10#1 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.C:\Dev\vcpkg\versions\v-\vcpkg-tool-lessmsi.json: note: consider removing the version constraint or choosing a value declared here
C:\Dev\vcpkg\ports\kubernetes\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\launch-darkly-server\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\libcurl-simple-https\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\libideviceactivation\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\libmediainfo\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\libmupdf\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\librdkafka\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\ports\libvault\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\libwandio\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\matroska\vcpkg.json: error: the "version>=" constraint to ebml names version 1.4.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\e-\ebml.json: note: consider removing the version constraint or choosing a value declared here
C:\Dev\vcpkg\ports\netcdf-c\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named dap
C:\Dev\vcpkg\ports\oatpp-curl\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\openjpeg\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named tools
C:\Dev\vcpkg\ports\openscap\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\opentelemetry-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\opentelemetry-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named otlp-http
C:\Dev\vcpkg\ports\opentelemetry-fluentd\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\openvino\vcpkg.json: error: the "version>=" constraint to protobuf names version 3.20.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\p-\protobuf.json: note: consider removing the version constraint or choosing a value declared here
note: the dependency is in the feature named onnx
C:\Dev\vcpkg\ports\openvino\vcpkg.json: error: the "version>=" constraint to protobuf names version 3.20.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\p-\protobuf.json: note: consider removing the version constraint or choosing a value declared here
note: the dependency is in the feature named onnx
C:\Dev\vcpkg\ports\openvino\vcpkg.json: error: the "version>=" constraint to protobuf names version 3.20.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\p-\protobuf.json: note: consider removing the version constraint or choosing a value declared here
note: the dependency is in the feature named paddle
C:\Dev\vcpkg\ports\openvino\vcpkg.json: error: the "version>=" constraint to protobuf names version 3.20.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\p-\protobuf.json: note: consider removing the version constraint or choosing a value declared here
note: the dependency is in the feature named paddle
C:\Dev\vcpkg\ports\openvino\vcpkg.json: error: the "version>=" constraint to protobuf names version 3.20.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\p-\protobuf.json: note: consider removing the version constraint or choosing a value declared here
note: the dependency is in the feature named tensorflow
C:\Dev\vcpkg\ports\openvino\vcpkg.json: error: the "version>=" constraint to protobuf names version 3.20.3 which does not exist in the version database. All versions must exist in the version database to be interpreted by vcpkg.
C:\Dev\vcpkg\versions\p-\protobuf.json: note: consider removing the version constraint or choosing a value declared here
note: the dependency is in the feature named tensorflow
C:\Dev\vcpkg\ports\osg\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named plugins
C:\Dev\vcpkg\ports\osg\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named tools
C:\Dev\vcpkg\ports\pdal\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\poppler\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\ports\ppconsul\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\proj\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named net
C:\Dev\vcpkg\ports\prometheus-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named push
C:\Dev\vcpkg\ports\qtvirtualkeyboard\vcpkg.json: error: the dependency t9write does not exist in the version database; does that port exist?
note: the dependency is in the feature named t9write
C:\Dev\vcpkg\ports\restclient-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\sentry-native\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named transport
C:\Dev\vcpkg\ports\shogun\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\tesseract\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\tgbot-cpp\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\ports\vsgxchange\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
note: the dependency is in the feature named curl
C:\Dev\vcpkg\ports\wxwidgets\vcpkg.json: error: the dependency curl does not exist in the version database; does that port exist?
C:\Dev\vcpkg\versions\z-\zlib.json: error: zlib@1.3#2 git tree 5ac18c6e6e3e2bf2a9e3d0bc8a845f198e4c4e05 does not match the port directory
C:\Dev\vcpkg\ports\zlib: note: the port directory has git tree a4344974697f4860565e4c5ba39d6b3e8086dbef
C:\Dev\vcpkg\ports\zlib\vcpkg.json: note: if zlib@1.3#2 is already published, update this file with a new version or port-version, commit it, then add the new version by running:
  vcpkg x-add-version zlib
  git add versions
  git commit -m "Update version database"
note: if zlib@1.3#2 is not yet published, overwrite the previous git tree by running:
  vcpkg x-add-version zlib --overwrite-version
  git add versions
  git commit -m "Update version database"
C:\Dev\vcpkg\versions\baseline.json: error: zlib is assigned 1.3, but the local port is 1.3#2
C:\Dev\vcpkg\ports\zlib\vcpkg.json: note: zlib is declared here
note: you can run the following commands to add the current version of zlib automatically:
  vcpkg x-add-version zlib
  git add versions
  git commit -m "Update version database"
```

### CODE REVIEW

**Comment by BillyONeal** [comment]
>C:\Dev\vcpkg\ports\azure-storage-cpp\vcpkg.json: error: the dependency cpprestsdk does not exist in the version database; does that port exist?

This is wrong, it is in the version database but the local port is damaged. Will fix...

File: src/vcpkg/paragraphs.cpp [hunk 1]
```diff
@@ -523,18 +602,16 @@ namespace vcpkg::Paragraphs
             auto maybe_port_location = (*port_entry)->get_version(*baseline_version);
             const auto port_location = maybe_port_location.get();
             if (!port_location) continue; // baseline version was not in version db (registry consistency issue)
-            auto maybe_spgh = try_load_port(fs, port_location->path);
+            auto maybe_spgh = try_load_port_required(fs, port_name, port_location->path).maybe_scfl;
```
Comment (BillyONeal) [inline src/vcpkg/paragraphs.cpp:605]: This needs to staple location back on?

**Review by JavierMatosD — APPROVED** [review]

File: src/vcpkg/commands.ci-verify-versions.cpp [hunk 2]
```diff
@@ -30,186 +29,450 @@ namespace
...
+                                          msg::path = scfl.port_directory())
+                                  .append_raw('\n')
+                                  .append_indent()
+                                  .append_raw("vcpkg x-add-version " + port_name + "\n")
+                                  .append_indent()
+                                  .append_raw("git add versions\n")
+                                  .append_indent()
+                                  .append(msgGitCommitUpdateVersionDatabase)
+                                  .append_raw('\n')
+                                  .append(msgVersionShaMismatch2,
```
Comment (BillyONeal) [inline src/vcpkg/commands.ci-verify-versions.cpp:242]: Audit for `If` vs `if`

File: azure-pipelines/end-to-end-tests-dir/ci-verify-versions.ps1 [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+$TestingRoot/ci-verify-versions-registry/ports/baseline-version-mismatch: message: baseline-version-mismatch@1.1 is correctly in the version database (cf8a1faa9f94f7ceb9513d65093d407e11ac1402)
+$TestingRoot/ci-verify-versions-registry/versions/baseline.json: error: baseline-version-mismatch is declared at version 1.0, but the local port is 1.1
+$TestingRoot/ci-verify-versions-registry/ports/baseline-version-mismatch/vcpkg.json: note: baseline-version-mismatch is declared here
+note: you can run the following commands to add the current version of baseline-version-mismatch automatically:
+  vcpkg x-add-version baseline-version-mismatch
+  git add versions
+  git commit -m `"Update version database`"
+$TestingRoot/ci-verify-versions-registry/ports/baseline-version-mismatch/vcpkg.json: All version constraints are consistent with the version database
+$TestingRoot/ci-verify-versions-registry/ports/baseline-version-missing: message: baseline-version-missing@1.0 is correctly in the version database (a5c21769008f52ed66afa344f13b786dde4b8d7d)
+$TestingRoot/ci-verify-versions-registry/versions/baseline.json: error: no version of baseline-version-missing is set
```
Comment (BillyONeal) [inline azure-pipelines/end-to-end-tests-dir/ci-verify-versions.ps1:36]:
```suggestion
$TestingRoot/ci-verify-versions-registry/versions/baseline.json: error: baseline-version-missing is not assigned a version
```

File: azure-pipelines/end-to-end-tests-dir/ci-verify-versions.ps1 [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  vcpkg x-add-version bad-git-tree --overwrite-version
+  git add versions
+  git commit -m `"Update version database`"
+$TestingRoot/ci-verify-versions-registry/versions/baseline.json: message: bad-git-tree@1.1 matches the current baseline
+$TestingRoot/ci-verify-versions-registry/ports/bad-git-tree/vcpkg.json: All version constraints are consistent with the version database
+$TestingRoot/ci-verify-versions-registry/ports/bad-history-name: message: bad-history-name@1.1 is correctly in the version database (f34f4ad3dfcc4d46d467d7b6aa04f9732a7951d6)
+$TestingRoot/ci-verify-versions-registry/versions/baseline.json: message: bad-history-name@1.1 matches the current baseline
+$TestingRoot/ci-verify-versions-registry/ports/bad-history-name/vcpkg.json: All version constraints are consistent with the version database
+$TestingRoot/ci-verify-versions-registry/ports/baseline-version-mismatch: message: baseline-version-mismatch@1.1 is correctly in the version database (cf8a1faa9f94f7ceb9513d65093d407e11ac1402)
+$TestingRoot/ci-verify-versions-registry/versions/baseline.json: error: baseline-version-mismatch is declared at version 1.0, but the local port is 1.1
```
Comment (BillyONeal) [inline azure-pipelines/end-to-end-tests-dir/ci-verify-versions.ps1:28]:
```suggestion
$TestingRoot/ci-verify-versions-registry/versions/baseline.json: error: baseline-version-mismatch assigned 1.0, but the local port is 1.1
```

**Review by JavierMatosD — DISMISSED** [review]
looks good to me :)

File: include/vcpkg/base/messages.h [hunk 5]
```diff
@@ -267,14 +267,4 @@ namespace vcpkg
 
 #include <vcpkg/base/message-data.inc.h>
 #undef DECLARE_MESSAGE
-
```
Comment (ras0219-msft) [inline include/vcpkg/base/messages.h:270]: Please separate this change.

File: src/vcpkg/paragraphs.cpp [hunk 6]
```diff
@@ -379,106 +407,146 @@ namespace vcpkg::Paragraphs
...
         {
-            const auto exists = ec != std::errc::no_such_file_or_directory;
-            if (exists)
+            auto manifest_exists = ec != std::errc::no_such_file_or_directory;
+            if (manifest_exists)
             {
-                return msg::format_error(msgFailedToParseManifest, msg::path = manifest_path)
-                    .append_raw("\n")
-                    .append(format_filesystem_call_error(ec, "read_contents", {manifest_path}));
+                return PortLoadResult{LocalizedString::from_raw(port_location.port_directory)
```
Comment (BillyONeal) [inline src/vcpkg/paragraphs.cpp:481]: @ras0219-msft suggests something like:

```
/long/path/to/my/zlib/abcd123515123123/vcpkg.json: error: permission denied
note: while loading port zlib
```

instead

**Comment by BillyONeal** [comment]
```
 version database to be interpreted by vcpkg.C:\Dev\vcpkg\versions\v-\vcpkg-tool-lessmsi.json
```

Check that this missing newline is fixed

File: src/vcpkg/paragraphs.cpp (hunk 6 above)
Comment (BillyONeal) [reply src/vcpkg/paragraphs.cpp:480]: Consider splitting this yet further to avoid printing the path in the error message part

**Comment by BillyONeal** [comment]
I marked this as draft for now while I'm still sucking review parts out

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**